capella_model_explorer/app.py

```diff
@@ -164,6 +164,22 @@ def rendered_report(template_id: str, model_element_uuid: str = "") -> t.Any:
             cls="prose svg-display dark:prose-invert",
         )
     )
+    return (
+        content,
+        fh.HttpHeader(k="Cache-Control", v="max-age=31536000"),
+        fh.HttpHeader(
+            k="Vary",
+            v=",".join(
+                [
+                    "model-explorer-version",
+                    "capellambse-version",
+                    "ctx-diags-version",
+                    "template-file-hash",
+                    "model-version",
+                ]
+            ),
+        ),
```

Comment:
Will this cause the cache to become invalid if new keys are added to Vary? If not, we should instead use a single header containing some sort of environment identifier / hash. New keys would then simply mean that the value of that identifier changes, thereby invalidating the cache as normal.

Reply:
I tried it out and it doesn't seem to invalidate. I suppose that does make sense, since the browser has a cached response already which doesn't vary based on the new header (which didn't exist yet), so why would it make a new request.

So, we should change this to a single header containing all the environment information that is relevant.

While at it, we should also mark it as a non-standard header with an `X-App-` prefix, e.g. `X-CME-Render-Environment: ...`.

Reply:
Looks good. That only leaves the issues with non-Git models.

Reply:
Interesting, and that isn't even a new one.

Also the other points:

> 1.  SHOULD assume that all parameters they create might become
>     standardized, public, commonly deployed, or usable across
>     multiple implementations.
>
> 2.  SHOULD employ meaningful parameter names that they have reason to
>     believe are currently unused.

Let's use "Render-Environment" instead then.

Reply:
That would go against the "might become [...] commonly deployed" point. Although it's "only" a cache tag in the end, there's no reason to arbitrarily limit the scope like this, we could in theory easily deploy the same technique in other webapps as well.

Reply:
This parameter which is very special to the present application will not become a standard. But, well, if you really dislike the `CME-` prefix, I can drop it.